services/payments/middleware.go

```diff
@@ -0,0 +1,41 @@
+package payments
+
+import (
+	"net/http"
+
+	"crypto"
+
+	appctx "github.com/brave-intl/bat-go/libs/context"
+
+	"github.com/brave-intl/bat-go/libs/httpsignature"
+	"github.com/brave-intl/bat-go/libs/middleware"
+)
+
+// ConfigurationMiddleware applies the current state of the service's configuration on the ctx
+func (service *Service) ConfigurationMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// wrap the request context in the baseCtx from the service
+		r = r.WithContext(appctx.Wrap(service.baseCtx, r.Context()))
+		next.ServeHTTP(w, r)
+	})
+}
+
+// AuthorizerSignedMiddleware requires that requests are signed by valid payment authorizers
+func (service *Service) AuthorizerSignedMiddleware() func(http.Handler) http.Handler {
+	authorizerVerifier := httpsignature.ParameterizedKeystoreVerifier{
+		SignatureParams: httpsignature.SignatureParams{
+			Algorithm: httpsignature.ED25519,
+			Headers: []string{
+				"(request-target)", "host", "date", "digest", "content-length", "content-type",
```

Comment:
The date header in the verifer has time constraints so messages that are at the back of a stream will expire and fail see [here](https://github.com/brave-intl/bat-go/blob/6164140dc823b61512cd57fa70d4faebc63cc5c1/libs/middleware/http_signed.go#L108)